### PR TITLE
libethdrivers/imx6: fix some bugs

### PR DESCRIPTION
**Pull request by axel-h**
This is the firsts step in a set of patches to improve the i.MX6 ethernet driver driver. I'm splitting this up into small chunks, so reviewing is simpler and merging should hopefully be quicker.

### CODE REVIEW

**Comment by axel-h** [comment]
- The license check fail comes from https://github.com/seL4/util_libs/issues/66
- the style checker is right, but in order to fix the bug, I will not fix the style everywhere for now. There are patched for this that will come later.

**Comment by lsf37** [comment]
Confirming that I'm happy for this to be merged with failing license check. I'm looking at the cause in a separate PR.

**Comment by axel-h** [comment]
@lsf37: nothing to investigate really here for you. The style checker is right to check every file that is touched by a PR for style violations. This is a common sloppy approach to get style fixed everywhere ("you touch it, you fix it") and usually that works out fine. Now what would be cool is a smarter style checker that can says that the change itself does not have style violations, but the file in general has some.

**Comment by lsf37** [comment]
> Now what would be cool is a smarter style checker that can say the code change itself does not have style violations, but the file in general has (always had) some.

Yes, that would be cool. There must be some way to do an intersection of diffs. I'll make a not of it, but I don't think I'll have time for it in the next few weeks. If someone else would want to look at this bit, I'd be happy to help with the GitHub part of the problem.

(What would be even cooler was if we could get the check to do direct code annotations -- unfortunately for GitHub Actions this is only possible for PRs made from within the repo, not without. To make this work universally, we'd have to write a GitHub web app and host that somewhere. Also doable, but more work.)

**Comment by axel-h** [comment]
I've fixed the style things also now.

**Comment by axel-h** [comment]
is anything preventing a merge? I have a bigger PR on top of this that will make the i.MX6 network driver work in QEMU...

**Comment by oliver-wm** [comment]
OK, will review this today!

**Comment by oliver-wm** [comment]
@ssrg-bamboo `test`

**Comment by ssrg-bamboo** [comment]
Hello, I'm a bot! I'll bring this PR into Trustworthy Systems and run some tests

**Comment by ssrg-bamboo** [comment]
All the tests we ran have passed! Nice job!

**Review by kent-mcleod — APPROVED** [review]

File: libplatsupport/src/plat/imx6/clock.c [hunk 1]
```diff
@@ -156,17 +157,18 @@ struct ccm_alg_regs {
     uint32_t  res8[4];
     /* MISC2 */
     alg_sct_t misc2;                 /* 0x170 */
-    uint32_t  res9[11];
-    /* USB phy control - Implemented here for the sake of
-     * componentisation since it shares the same register space */
+    uint32_t  res9[8];
```
Comment (oliver-wm) [inline libplatsupport/src/plat/imx6/clock.c:160]: Will reducing this to 8 change the addresses of the phy1 phy2 regs below? What was the reason for this?

**Review by oliver-wm — APPROVED** [review]

**Comment by oliver-wm** [comment]
@ssrg-bamboo `test`

**Comment by ssrg-bamboo** [comment]
Hello, I'm a bot! I'll bring this PR into Trustworthy Systems and run some tests

File: libplatsupport/src/plat/imx6/clock.c (hunk 1 above)
Comment (axel-h) [reply libplatsupport/src/plat/imx6/clock.c:160]: See the comment with the address. 11 is clearly wrong here. `alg_sct_t` is 16 byte. So 0x170 + 16 + 4*8 = 0x1a0. I thin the 11 came from the assumption that `alg_sct_t` is 4 bytes only- Or even from before `alg_sct_t` existed and the was simply a uint32.

Comment (oliver-wm) [reply libplatsupport/src/plat/imx6/clock.c:160]: Ah cool! Good spot. Have these changes been tested?

Comment (axel-h) [reply libplatsupport/src/plat/imx6/clock.c:160]: I'm using these changes for a few months in my QEMU NIC support branch now. However, I have not been using the USB PHY so far. This change specifically comes purely from code/datasheet reviews.

**Comment by oliver-wm** [comment]
Ok this is merged now.

**Pull request closed without merging**